ui-discover/src/main/java/app/tivi/home/discover/Discover.kt

```diff
@@ -0,0 +1,424 @@
+
+            Column(Modifier.gravity(Alignment.CenterVertically)) {
+                val textCreator = DiscoverTextCreatorAmbient.current
+                ProvideEmphasis(EmphasisAmbient.current.disabled) {
+                    Text(
+                        text = textCreator.seasonEpisodeTitleText(season, episode),
+                        style = MaterialTheme.typography.caption
+                    )
+                }
+
+                Spacer(Modifier.preferredHeight(4.dp))
+
+                ProvideEmphasis(EmphasisAmbient.current.high) {
+                    Text(
+                        text = episode.title
+                            ?: stringResource(R.string.episode_title_fallback, episode.number!!),
+                        style = MaterialTheme.typography.body1
+                    )
+                }
+            }
+        }
+    }
+}
+
+@Composable
+private fun <T : EntryWithShow<*>> CarouselWithHeader(
+    items: List<T>,
+    title: String,
+    refreshing: Boolean,
+    onItemClick: (TiviShow) -> Unit,
+    onMoreClick: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    Column(modifier) {
+        if (refreshing || items.isNotEmpty()) {
+            Spacer(Modifier.preferredHeight(16.dp))
+
+            Header(
+                title = title,
+                loading = refreshing,
+                action = {
+                    TextButton(
+                        onClick = onMoreClick,
+                        contentColor = MaterialTheme.colors.secondary,
+                        modifier = Modifier.alignWithSiblings(FirstBaseline)
+                    ) {
+                        Text(text = stringResource(R.string.header_more))
+                    }
+                },
+                modifier = Modifier.fillMaxWidth()
+            )
+        }
+        if (items.isNotEmpty()) {
+            EntryShowCarousel(
+                items = items,
+                onItemClick = onItemClick,
+                modifier = Modifier.preferredHeight(192.dp).fillMaxWidth()
+            )
+        } else {
+            // TODO empty state
+        }
+    }
+}
+
+@Composable
+private fun <T : EntryWithShow<*>> EntryShowCarousel(
+    items: List<T>,
+    onItemClick: (TiviShow) -> Unit,
+    modifier: Modifier = Modifier
+) {
+    Carousel(
+        items = items,
+        contentPadding = InnerPadding(start = 16.dp, top = 8.dp, end = 16.dp, bottom = 8.dp),
+        itemSpacing = 4.dp,
+        modifier = modifier
+    ) { item, padding ->
+        PosterCard(
+            show = item.show,
+            poster = item.poster,
+            onClick = { onItemClick(item.show) },
+            modifier = Modifier
+                .padding(padding)
+                .fillParentMaxHeight()
+                .aspectRatio(2 / 3f)
+        )
+    }
+}
+
+@OptIn(ExperimentalAnimationApi::class)
+@Composable
+private fun Header(
+    title: String,
+    loading: Boolean = false,
+    action: (@Composable () -> Unit)? = null,
```

Comment:
Make last param to enable trailing lambda.

Reply:
Done